device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -75,16 +77,13 @@
     private long receivePeriodInMilliseconds;
 
     private IotHubTransport transport;
-    private DeviceClientConfig config;
     private IotHubSendTask sendTask = null;
     private IotHubReceiveTask receiveTask = null;
-    private IotHubClientProtocol protocol = null;
 
     private ScheduledExecutorService receiveTaskScheduler;
     private ScheduledExecutorService sendTaskScheduler;
     private IotHubConnectionStatus state;
 
-    private List<DeviceClientConfig> deviceClientConfigs = new LinkedList<>();
```

Comment:
There really wasn't a good reason to have a list of configs or the protocol at this level. I've delegated that responsibility down to the transport connection since that layer actually needs the protocol value and configs